apps/kg/models/mxnet/tensor_models.py

```diff
@@ -71,9 +71,9 @@ def update(self):
                 grad_sum = grad_sum.as_in_context(ctx)
             self.state_sum[grad_indices] += grad_sum
             std = self.state_sum[grad_indices]  # _sparse_mask
-            std_values = nd.expand_dims(nd.sqrt(std) + 1e-10, 1)
-            if self.gpu >= 0:
-                std_values = std_values.as_in_context(mx.gpu(self.args.gpu))
+            if gpu_id >= 0:
+                std = std.as_in_context(mx.gpu(gpu_id))
+                std_values = nd.expand_dims(nd.sqrt(std) + 1e-10, 1)
```

Comment:
why do you move the code here? shouldn't we call sqrt and expand_dims regardless of the device?